src/Modules/AdminDashboard/Componenets/Ads/AdsList/AdsList.tsx

```diff
@@ -1,3 +1,566 @@
+import {
+  adsForm,
+  AdsInterface,
+  roomsInterface,
+  UpdateAdsForm,
+} from "../../../../../Interfaces/interFaces";
+import {
+  Box,
+  Grid,
+  Typography,
+  Button,
+  Modal,
+  Backdrop,
+  Fade,
+  CircularProgress,
+  TextField,
+  Select,
+  InputLabel,
+  MenuItem,
+  SelectChangeEvent,
+  FormControl,
+} from "@mui/material";
+import MUIDataTable from "mui-datatables";
+import { CacheProvider } from "@emotion/react";
+import createCache from "@emotion/cache";
+import { DeleteForever, Draw, HighlightOff } from "@mui/icons-material";
+import { useForm, Controller } from "react-hook-form";
+import { toast } from "react-toastify";
+import delImg from "../../../../../assets/images/noData.png";
+import { getBaseUrl } from "../../../../../Utils/Utils";
+import { useAuth } from "../../../../../Context/AuthContext/AuthContext";
+
+const muiCache = createCache({
+  key: "mui-datatables",
+  prepend: true,
+});
+
 export default function AdsList() {
-  return <div>AdsList</div>;
+  const [ads, setAds] = useState<AdsInterface[]>([]);
+  const [rooms, setRooms] = useState([]);
+  const [open, setOpen] = useState(false);
+  const [openDelete, setOpenDelete] = useState(false);
+  const [spinner, setSpinner] = useState<boolean>(false);
+  const [adID, setAdID] = useState<string | null>(null);
+  const [adName, setAdName] = useState<string | null>(null);
+  const [isUpdate, setIsUpdate] = useState<boolean>(false);
+  const [roomSelect, setRoomSelect] = useState("");
+  const [isActiveSelect, setIsActiveSelect] = useState<string | boolean>("");
+  const [totalCount, setTotalCount] = useState(10);
+
+  const {
+    handleSubmit,
+    control,
+    setValue,
+    formState: { errors },
+  } = useForm<adsForm, UpdateAdsForm>();
+  const { requestHeaders } = useAuth();
+  const handleRoomChange = (event: SelectChangeEvent) => {
+    setRoomSelect(event.target.value as string);
+  };
+
+  const handleActiveChange = (event: SelectChangeEvent) => {
+    const isActive = event.target.value === "true";
+    setIsActiveSelect(isActive);
+  };
+
+  const handleOpen = () => setOpen(true);
+  const handleOpenDelete = () => setOpenDelete(true);
+  const handleCloseDelete = () => setOpenDelete(false);
+  const handleClose = () => {
+    setOpen(false);
+    setAdID(null);
+    setAdName(null);
+    setValue("room", "");
+    setValue("discount", "");
+    setValue("isActive", false);
+    setIsUpdate(false);
+    setRoomSelect("");
+    setIsActiveSelect("");
+  };
+
+  const handleUpdate = (value: AdsInterface) => {
+    setAdID(value._id);
+    setAdName(value.room.roomNumber);
+    handleOpen();
+    setValue("discount", value.room.discount);
+    setValue("isActive", value.isActive);
+    setIsActiveSelect(String(value.isActive));
+    setIsUpdate(true);
+  };
+
+  const handleDelete = (value: AdsInterface) => {
+    setAdID(value._id);
+    setAdName(value.room.roomNumber);
+    handleOpenDelete();
+  };
+
+  const columns = [
```

Comment:
I prefer to make this representation data at the bottom and make the code at the bottom for better readability.